src/main/java/core/basesyntax/bookstore/controller/CartController.java

```diff
@@ -0,0 +1,136 @@
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/cart")
+@Tag(name = "Shopping cart management",
+        description = "Endpoints to manage shopping cart for currently logged in user")
+@RequiredArgsConstructor
+public class CartController {
+    private final CartService cartService;
+
+    @Operation(summary = "Get shopping cart")
+    @ApiResponse(
+            responseCode = "200",
+            description = "Receive shopping cart info and its' items"
+    )
+    @PreAuthorize("hasRole('USER')")
+    @GetMapping
+    public ShoppingCartDto getShoppingCart(Authentication authentication) {
+        User user = (User) authentication.getPrincipal();
+        return cartService.getShoppingCart(user);
+    }
+
+    @Operation(summary = "Create cart item")
+    @ApiResponses({
+            @ApiResponse(
+                    responseCode = "200",
+                    description = "Created cart item",
+                    content = @Content(schema =
+                    @Schema(implementation = CartItemWithoutBookDto.class))
+            ),
+            @ApiResponse(
+                    responseCode = "400",
+                    description = "Validation failed",
+                    content = @Content(schema = @Schema(
+                            implementation = ValidationExceptionDto.class
+                    ))
+            ),
+            @ApiResponse(
+                    responseCode = "404",
+                    description = "Didn't find the book by id",
+                    content = @Content(schema = @Schema(implementation = ExceptionDto.class))
+            )
+    })
+    @PreAuthorize("hasRole('USER')")
+    @PostMapping
+    public CartItemWithoutBookDto addItemToShoppingCart(@RequestBody
+                                                        @Valid
+                                                        AddCardItemRequestDto requestDto,
+                                                        Authentication authentication) {
```

Comment:
```suggestion
    public CartItemWithoutBookDto addItemToShoppingCart(
        @RequestBody @Valid AddCardItemRequestDto requestDto,
        Authentication authentication
    ) {
```
I recommend using this formatting. As for me, it’s  more readable